### DISCUSSION TITLE
How to use "turn order"-feature with to-changing spells/gems

### DISCUSSION THREAD

**Discussion by Vortilion** [on3iro/aeons-end-randomizer#430] [Q&A]
Hi,

Im just starting with AE and found your react-App for it. Great effort, thanks for that! Was also thinking about implementing sth. similar with Angular before I found your App. ;) My question is: how do I use the turn order feature correctly and what are the buttons on the drawn players for? I tried playing a game with the mages Lash an Jian. When playing Lash‘ Quartz Shard I was struggling using your App with it. How would I use Quartz Shard?

> Reveal the top card of the turn order deck. You may place that card on the bottom of the turn order deck; otherwise return it to the top.

another question, looks like a bug to me: if I install it as a PWA via „add to homescreen“ it stops working. The spinner is visible but the rest won’t appear?

**Comment by on3iro** [comment]
Hi @Vortilion,

regarding your bug: unfortunately I can't reproduce this (tested on various devices - it works fine here). Maybe try closing the app after adding it to your homescreen and first opening it. It could be that the services worker has had some difficulties our something.
What browser are you using?

Regarding the turn order feature:
There are three buttons on each turn order card.
The upward arrow means: "put the card on top of the turn order deck".
The downward arrow means: "put the card at the bottom of the to deck".
The circling arrows mean: "shuffle the card into the turn order deck".

So when using Quartz shard you would simply draw the next turn order card and then decide to either use the upward or downward arrow. (so to be clear: we decided against adding a "reveal" feature, as drawing basically does the same thing. Just imagine how you would do it, when you would be using the physical TO-deck ;) )

Hope that helps :)

**Comment by Vortilion** [comment]
Hi Theo, ich sehe gerade, dass deutsch wohl auch geht? ;-) Ich versuchs mal so... Danke für die schnelle Antwort! Neustart der App habe ich schon versucht, leider tritt das Problem jedesmal auf, auch auf unterschiedlichen Geräten (iOs und iPadOS, also Safari als Browser). ServiceWorker hatte ich auch schon im Verdacht, leider habe ich keine Möglichkeit, das zu debuggen (keinen Mac-Rechner). Wenn ich hier ein Video einstellen kann zu Vorführungszwecken versuche ich das gleich mal.

Danke für die Erklärung der Funktionen! Entweder ich bin blind oder ich habe keine "Anleitung" für die App gefunden... Vielleicht bin ich auch nur zu neu im Spiel, so dass es für mich nicht intuitiv in der Bedienung war. Aber bin auch kein UX'ler... :-p

**Reply by Vortilion** [reply]
Unfassbar. Wie es manchmal so ist scheint es jetzt auf einmal zu funktionieren. Als Entwickler muss ich aber sagen, dass ich sich-selbst-lösenden-Bugs nicht traue... ;-)

**Reply by on3iro** [reply]
Haha, ja geht mir ähnlich. Wobei ich bei allem was PWAs und insbesondere Service Worker betrifft inzwischen gezwungenermaßen etwas entspannter bin, da die gern mal nicht so 100%ig tun, was man erwartet, relativ schwer zu debuggen sind und sich die Probleme fast immer wie von Zauberhand von allein lösen ;)

Bzgl. der UX:
Bisher haben wir, was das TO-Feature betrifft kein negatives Feedback bekommen. Der Teil der App hat sich fuer uns relativ intuitiv angefuehlt. Wir haben daher erstmal nur unser komplexestes Feature mit user docs versehen. Gerade habe ich auch leider sehr wenig Zeit, um an der App zu arbeiten, da wir frisch Eltern geworden sind. Wir haben auch bereits die Files fuer die Uebersetzung der zweiten Welle ins Deutsche, aber schlicht noch keine Zeit gehabt, die Texte in unsere App einzupflegen.

Solltest du noch irgendwo Fragen haben, melde dich einfach. Ich versuche  dann zeitnah zu antworten. :)

ps: hier  das, was an "user" docs existiert https://on3iro.github.io/aeons-end-randomizer/#/

**Reply by Vortilion** [reply]
Hey, ich wäre jetzt auch nicht so weit gegangen von "negativem Feedback" zu sprechen, so war das nicht gemeint. Es war nur für mich persönlich nicht intuitiv, da es auch keinerlei UX-Feedback gibt, wenn man einen der Button drückt, deshalb wusste ich nicht, was genau da passiert, wenn die TO-Karte plötzlich verschwindet. ;-)

Was PWAs angeht bin ich selbst leider nicht net ganz so fit, habe bisher nur ein wenig mit dem Manifest und ServiceWorkern gearbeitet, ist aber schon ne Weile her. Und React ist für mich auch noch Neuland, bin eher bei Google's Angular unterwegs. React steht aber auf meiner Liste... Danke Dir auf jeden Fall für die raschen Antworten, und Glückwunsch zur Familien-Deluxe-Erweiterung! ;-) Unsere Kleine ist nun 4.

**Reply by Vortilion** [reply]
> ps: hier das, was an "user" docs existiert https://on3iro.github.io/aeons-end-randomizer/#/

Hm, vielleicht komme ich darauf zurück und stelle einen Pull Request ein, zumindest, was das TO-Feature angeht. :-D Danke für den Hinweis!
